eng/Versions.props

```diff
@@ -120,7 +120,8 @@
   </PropertyGroup>
   <PropertyGroup>
     <!-- Dependencies from https://github.com/NuGet/NuGet.Client -->
-    <NuGetVersioningPackageVersion>5.8.0</NuGetVersioningPackageVersion>
+    <NuGetPackagingPackageVersion>6.7.0-preview.2.50</NuGetPackagingPackageVersion>
```

Comment:
@dsplaisted, @mmitche - Can we upgrade this nuget dependency to be a the latest version?  Installer was declaring a nuget version.details.xml dependency but it was not used here.

Reply:
It should be coherent with what dotnet/sdk has. It currently matches https://github.com/dotnet/sdk/blob/main/eng/Version.Details.xml#L115-L118

Reply:
It looks like this is the latest in the VS 17.7 channel, which is what main is currently tracking.

Reply:
Right - the versions.details.xml dependency already existing in installer and was set to be coherent with the sdk.

Reply:
> Also, you will need to ensure that the two NuGet dependencies exist in dotnet/sdk's Version.Details.xml file. Otherwise strict coherency will fail.

Can you help me understand why?  That makes me question if I did the right thing in this PR?

Reply:
Per our offline discussion, I reverted the change to add new Version.Details.xml dependencies and updated the Versions.props to use the existing NuGet dependency.

Reply:
@mmitche please review to ensure I updated as you expected.